revscoring/languages/persian.py

```diff
@@ -5,57 +5,78 @@
-    r"شَ?هید\b", r"آزادهٔ? سرا?فراز",
-    r"شادروان", r"جهانخوار", r"مستکبر", r"ملحد", r"ملعون", r"عل[يی]ه‌? ?السلام",
+    # Partial words
+    r"آله", r"فرموده?", r"فرمودند", r"السلام", r"حضرت\b", r"\([سعص]\)",
+    r"\(قس\)",  r"آزادهٔ? سرا?فراز",  r"جناب",  r"قدس‌? ?سره‌? ?شریف",
+    r"\(عج\)", r"\bامام\b", r"فرمودند", r"روحی?‌? ?ا?ل?فدا", r"شَ?هید\b",
+    r"شادروان", r"جهانخوار", r"مستکبر", r"ملحد", r"ملعون", r"\bشهادت",
     r"(لعن[تة]|رحمت|صلی|صلوات|سلام)‌? ?اللہ", r"دام‌? ?(ظلّ?ه|برکات)",
-    r"قدس‌? ?سره‌? ?شریف",
-    r"اسقف محترم", r"خدا[یش]? بیامرز", r"دار فانی", r"به هلاکت",
-    r"سَقط شد", r"ا?علیا?‌? ?حضرت",
+    r"اسقف محترم", r"خدا[یش]? بیامرز", r"دار فانی", r"سَقط شد",
     r"خادم خادمان", r"مقام معظّ?م", r"(حرم|مرقد) مطهر", r"\bمرحوم\b",
-    r"\bشهادت", r"شاهنشاه\b",
-    r"علیها", r"مد ?ظله"
+    r"علیها", r"مد ?ظله", r"ایشان", r"منظورشان", r"ماشالا", r"\(ره\)",
+    r"به هلاکت", r"عل[يی]ه‌? ?السلام", r"شاهنشاه\b", r"ا?علیا?‌? ?حضرت",
+    # Slangy verbs
+    r"\b(ن?می‌? ?|ب|ن)([یا]فشون|پاشون|پرورون|پرون|پوسون|پوشون|پیچون|" +
+    r"تابون|تازون|ترسون|ترکون|تکون|تون|جنبون|جوشون|چپون|چربون|چرخون|" +
+    r"چرون|چسبون|چشون|چکون|چلون|خارون|خراشون|خشکون|خندون|خوابون|" +
+    r"خورون|خون|خیسون|درخشون|رسون|رقصون|رنجون|رون|سابون|ستون|" +
+    r"سوزون|ش|شورون|غلتون|فهمون|کوبون|گذرون|گردون|گریون|گزین|گسترون|" +
+    r"گ|گنجون|لرزون|لغزون|لمبون|مالون|ا?نداز|نشون|هراسون|وزون)" +
+    r"(م|ی|ه|یم|ید|ن)\b",
+    r"\b(ن?می‌? ?|ب)(شین|مون)(م|ی|ه|یم|ید|ن)\b",
+    r"\b(ن?می‌? ?|ب|ن)(چا|خا|خوا)(م|ی|د|یم|ید|ن)\b",
+    r"\b(ن?می‌? ?|ب|ن)([یا]رز|[یا]فت|[یا]فراز|[یا]فروز|[یا]ندیش|[یا]نگیز|ایست|" +
+    r"ی?[اآ]زار|ی?[اآ]فرین|ی?[اآ]مرز|ی?[اآ]میز|ی?[اآ]ور|ی?[اآ]ویز|بار|باز|" +
+    r"باف|بال|بخش|بر|بلع|بند|بوس|بین|پاش|پذیر|پر|پراکن|پرداز|پرس|پرست|پرور|" +
+    r"پز|پسند|پلک|پندار|پوس|پوش|پیچ|پیوند|تاب|تپ|تراش|ترس|ترش|ترک|جنب|جنگ|جه|" +
+    r"جو|جوش|چاپ|چپ|چر|چرب|چرخ|چسب|چش|چک|چین|خار|خر|خز|خشک|خند|خواب|خور|دار|" +
+    r"در|درخش|دزد|دم|دوز|دوش|رس|رقص|رنج|ریز|رین|زار|زن|ساب|ساز|سپار|سنج|شاش|" +
+    r"شکاف|شکف|شکن|شمر|شناس|شنو|شور|طپ|طلب|غر|غلت|فرست|فروش|فریب|فهم|قاپ|کار|" +
+    r"کش|کن|کوب|گذار|گذر|گرد|گری|گریز|گستر|گمار|گند|گوز|گیر|لرز|لغز|لنگ|لول|لیس|" +
+    r"ماس|مال|مک|میر|ناز|نال|نگر|نواز|نورد|نویس|هراس|ورز|وز|یاب)(ه|ن)\b",
+    r"\b(ن?می‌? ?|ب|ن)(پا|کاه|گا)ن\b",
+    # Slangy adverbs or words
+    r"\b(ازش|انگار|اونه?ا|ایشون|اینجوری?|این[وه]|بازم|باهاش|براتون|برام|برای[مت]ا?ن?|" +
+    r"به[مش]|بی‌خیال|تموم|چ?جوری|چیه|دیگه|کدوم|مونده)\b",
+    # Salutation words
+    r"\b(ارادتمندم?|ببخشید|بفرما|جناب|خدمت[تم][وا]?ن?|خوبی|خودت[وا]ن|خودم[وا]ن|خوشحالم|" +
+    r"خوشم|درود|دوستدار|سپاسگزارم|سلام|شرمنده|شماست|فلانی|مبارک|متشکرم|محترمت[وا]ن|" +
+    r"مخالفم|مرسی|ممنونم?|منظورت[وا]ن|موافقم|نظر[تم][وا]?ن?|نظرم|یادت[وا]ن)\b",
+    # wikipedia words
+    r"\b(بحث\:|بحثم|کاربر\:|میان‌ویکی|ویکی‌? ?فا)\b"
 ]
 
 badwords = [
+    # Fingilish bad words
     r"(madar|nanae|zan|khahar)\s*?(ghahbeh|ghahveh|ghabe|jendeh?|be khata)",
```

Comment:
Regretfully, we can't support spaces in words right now.  See https://github.com/wiki-ai/revscoring/issues/164

If there's some way we can record these multi-word badwords for future use, but only include single-word badwords right now, that would be most desirable.